odoo/models.py

```diff
@@ -67,8 +67,9 @@
 _logger = logging.getLogger(__name__)
 _unlink = logging.getLogger(__name__ + '.unlink')
 
-regex_order = re.compile(r'^(\s*([a-z0-9:_]+|"[a-z0-9:_]+")(\.id)?(\s+(desc|asc))?\s*(,|$))+(?<!,)$', re.I)
+regex_order = re.compile(r'^(\s*([a-z0-9:_]+|"[a-z0-9:_]+")(\.[a-z0-9]+)?(\s+(desc|asc))?\s*(,|$))+(?<!,)$', re.I)
 regex_object_name = re.compile(r'^[a-z0-9_.]+$')
+regex_alphanumeric = re.compile(r'^[a-z0-9]+$')
```

Comment:
[isalnum](https://docs.python.org/3/library/stdtypes.html#str.isalnum) Maybe ?

Reply:
Hum, not excatly the same, those character will be allowed :/

https://en.wikipedia.org/wiki/Eastern_Arabic_numerals

![screenshot_spect](https://user-images.githubusercontent.com/54316226/203286564-1ef019bd-a91b-45de-9085-c93bdb24fa66.png)